src/data.rs

```diff
@@ -34,11 +36,28 @@ pub struct Data {
     inner: Vec<u8>,
 }
 
+/// An error indicating a string was not valid XML data.
+#[derive(Debug)]
+pub struct InvalidXmlData(base64::DecodeError);
```

Comment:
Does this need `#[non_exhaustive]` or is the fact that `base64::DecodeError` is not `pub` sufficient?

Reply:
Assuming we're using `#[non_exhaustive]` to prevent construction of this type from outside the crate, then the private internal field should accomplish the same thing. :)

Reply:
I was thinking more about pattern matching against the number of tuple elements, even thought you can't extract the value i.e.
```rust
let InvalidXmlData(_) = err;
```
I think that's forbidden by privacy rules. I'm a bit out of the habit of writing Rust.